stdlib/public/Reflection/TypeRefBuilder.cpp

```diff
@@ -193,36 +193,85 @@ const TypeRef *TypeRefBuilder::lookupSuperclass(const TypeRef *TR) {
   return Unsubstituted->subst(*this, *SubstMap);
 }
 
+static llvm::Optional<StringRef> FindOutermostModuleName(NodePointer Node) {
+  // Breadth first search until we find the module name so we find the outermost
+  // one.
+  llvm::SmallVector<NodePointer, 8> queue;
+  queue.push_back(Node);
+  // Instead of removing items from the front of the queue we just iterate over
+  // them.
+  for (size_t i = 0; i < queue.size(); ++i) {
+    NodePointer Current = queue[i];
+    queue.erase(queue.begin());
```

Comment:
To add to Mike's comment, it looks like this will skip nodes.

```
queue = [Node]
current = queue[0]
queue.erase(0)
queue.push_back(...)
---
queue = [...]
current = queue[1] // skips [0]
queue.erase(0) // deletes [0]
queue.push_back(...)
---
queue = [...]
current = queue[2] // skips [0, 1]
queue.erase(0) // deletes [0]
queue.push_back(...)
```

Reply:
Yeah that's a mistake, I changed it right before committing because I thought it'd be more efficient to not remove nodes but forgot to remove that line 😬